Assets/MixedRealityToolkit.Extensions/Sharing/Localization/Extensions.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Numerics;
+using System.Threading;
+using System.Threading.Tasks;
+using UQuaternion = UnityEngine.Quaternion;
+using UVector3 = UnityEngine.Vector3;
+
+namespace Microsoft.MixedReality.Toolkit.Extensions.Experimental.Sharing
+{
+    /// <summary>
+    /// Helper Unity side extensions to enable the core abstraction to be Unity agnostic.
+    /// </summary>
+    public static class Extensions
+    {
+        /// <summary>
+        /// Converts a <see cref="Vector3"/> to a <see cref="UVector3"/>.
+        /// </summary>
+        public static UVector3 AsUnityVector(this Vector3 input) => new UVector3(input.X, input.Y, input.Z);
+
```

Comment:
How hard do we feel a migration to System.Numerics types will be in the future if we only used Unity types for the time being?

Non-Unity middleware may be unable to play well with .Net types, which I think is the main intended worth in this change. It looks like this namespace exists in .NET 4.0+ frameworks. I don't know if anyone uses .NET frameworks older than 4.0, but I don't think this would resolve correctly in this current form for those that do. I think I have seen some issues related to this based on folks using the HoloToolkit with older Unity/.NET versions. Does the consumer confusion that could be generated with this change outweigh the worth in standardizing on a non-Unity type?

From what I understand the intention here is to build a more universal experience by using .Net types. Although these extension methods are helpful, I think that most Unity developers will still prefer Unity types. I think for the time being I would argue that we should continue to use Unity types. Once we have a stronger perspective on how to build components compatible with Unreal and Unity, it may be a better time to try and introduce these sorts of type conversions.

Reply:
I am choosing to go this route for several reasons:
- Feel the pain of going this route, so that when we do make a full choice we understand. (I will be documenting this)
- It is simple to flip back to only Unity types, and we are in the Experimental branch anyways.
- Experiment with having this code used in Unreal.

---
In reply to: [270518326](https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/3732#discussion_r270518326) [](ancestors = 270518326)